uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveBalanceChange.php

```diff
@@ -76,8 +86,29 @@ public static function getBalanceForEntitlement(LeavePeriodEntitlement $periodEn
       LEFT JOIN {$leaveRequestTable} leave_request 
              ON leave_request_date.leave_request_id = leave_request.id AND
                 leave_request.is_deleted = 0
+      LEFT JOIN {$contractTable} contract 
```

Comment:
Not in this case. The problem here is that we have to join with balance changes that are either linked to a leave request date or to an entitlement. On `getLeaveRequestBalanceForEntitlement` we only care about those linked to the dates of a leave request, hence a INNER JOIN is possible